accord-core/src/main/java/accord/primitives/KeyDeps.java

```diff
@@ -428,13 +439,18 @@ public List<TxnId> txnIds(Key key)
         if (keyIndex < 0)
             return Collections.emptyList();
 
+        return txnIdsForKeyIndex(keyIndex);
+    }
+
+    public SortedRelationList<TxnId> txnIdsForKeyIndex(int keyIndex)
+    {
+        int[] keysToTxnIds = keysToTxnIds();
         int start = startOffset(keyIndex);
         int end = endOffset(keyIndex);
-        int size = end - start;
-        return txnIds(keysToTxnIds, start, size);
+        return txnIds(keysToTxnIds, start, end);
     }
 
-    public List<TxnId> txnIds(Range range)
+    public SortedRelationList<TxnId> txnIds(Range range)
```

Comment:
```suggestion
    @SuppressWarnings("unchecked")
    public SortedRelationList<TxnId> txnIds(Range range)
```

This is needed due to `SortedRelationList.EMPTY;`... if you create a `empty()` method you can push this suppress warning there and avoid 3 compiler warnings